app/lib/common/widgets/attachments/post_attachment_selection.dart

```diff
@@ -0,0 +1,185 @@
+    this.convo,
+  });
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() =>
+      _PostAttachmentSelectionState();
+}
+
+class _PostAttachmentSelectionState
+    extends ConsumerState<PostAttachmentSelection> {
+  @override
+  Widget build(BuildContext context) {
+    final files = widget.files;
+    final titleTextStyle = Theme.of(context).textTheme.titleSmall;
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.start,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(12),
+          child: Text(
+            'Attachments Selected (${files.length})',
+            style: titleTextStyle,
+          ),
+        ),
+        const SizedBox(height: 15),
+        _buildSelectedDrafts(),
+        const Spacer(),
+        _buildActionButtons(),
+      ],
+    );
+  }
+
+  Widget _buildSelectedDrafts() => Padding(
+        padding: const EdgeInsets.all(12),
+        child: Wrap(
+          spacing: 5.0,
+          runSpacing: 10.0,
+          children: List.generate(
+            widget.files.length,
+            (idx) => AttachmentDraftItem(file: widget.files[idx]),
+          ),
+        ),
+      );
+
+  // send/cancel buttons
+  Widget _buildActionButtons() {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 10),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceAround,
+        children: <Widget>[
+          OutlinedButton(
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+            child: const Text('Cancel'),
+          ),
+          ElevatedButton(
+            onPressed: () async {
+              // we are popping out twice to clear selection sheet too!
+              Navigator.of(context).pop();
+              if (widget.convo != null) {
+                handleChatAttachmentSend(widget.files);
+              } else if (widget.manager != null) {
+                handleAttachmentSend();
+              }
+              // manager not present, this shouldn't lead up here
+              return;
+            },
+            child: const Text('Send'),
+          ),
+        ],
+      ),
+    );
+  }
+
+  // if generic attachment, send via manager
+  void handleAttachmentSend() async {
+    final attachmentDraftsNotifier =
+        ref.read(attachmentDraftsProvider(widget.manager!).notifier);
+    for (var file in widget.files) {
+      await attachmentDraftsNotifier.addDraft(file);
+    }
+    await attachmentDraftsNotifier.sendDrafts();
+  }
+
+  // if room attachment, send via timeline stream message
+  void handleChatAttachmentSend(
+    List<File> selectedAttachments,
+  ) async {
+    final roomId = widget.convo!.getRoomIdStr();
+    final client = ref.read(alwaysClientProvider);
+    final inputState = ref.read(chatInputProvider(roomId));
+    final stream = widget.convo!.timelineStream();
+
+    try {
+      for (var file in selectedAttachments) {
```

Comment:
This was clearly different before at least for chat and should be done in a similar way for attachments: instead of generalizing everything into files and then guess the content via a mime-type as we do here, before when a user selected an image, we sent it as an image, regardless of the internal types. We should do that, in particular because if the user selected something to be sent as a file or video or whatnot, it isn't our place to just replace that semantic information with something else we _think_ we detected. If it was selected as a file, we should sent it as a file, same for video and image... 

I understand this isn't a degression from before, so I won't consider this a blocker. But if we merge it this way, let's at least put up an issue asking for a refactor of this into the proper types provided by the user...

Reply:
Fixed by c53e1ac